piccolo/testing/random_builder.py

```diff
@@ -43,12 +81,18 @@ def next_enum(cls, e: t.Type[enum.Enum]) -> t.Any:
     def next_float(cls, minimum=0, maximum=2147483647, scale=5) -> float:
         return round(random.uniform(minimum, maximum), scale)
 
+    @classmethod
+    def next_decimal(cls, digits: t.Tuple[int, int] | None = (4, 2)) -> float:
+        precision, scale = digits or (4, 2)
+        return cls.next_float(maximum=10 ** (precision - scale), scale=scale)
```

Comment:
Could this be related to the precision of the Decimal type? For instance:
```python=
>>> Decimal(100.1)
Decimal('100.099999999999994315658113919198513031005859375')
```

Reply:
Good point. Decimals are so weird. I wonder if instead we can create a decimal using a string.

Some pseudo code, where we get two random integers - one for before the decimal point, and one for after the decimal point:

```python
decimal.Decimal(f'{get_some_number()}.{get_some_other_number()}')
```

Rather than passing a float to `Decimal`.

Reply:
Fantastic suggestion, indeed!

Reply:
Maybe like:
```python
class RandomBuilder:
    ...
    @classmethod
    def next_float(cls, minimum=0, maximum=2147483647, scale=5) -> float:
        return round(random.uniform(minimum, maximum), scale)

    @classmethod
    def next_decimal(
        cls, precision: int = 4, scale: int = 2
    ) -> decimal.Decimal:
        float_number = cls.next_float(
            maximum=10 ** (precision - scale), scale=scale
        )
        return decimal.Decimal(str(float_number))
```